Source/Libraries/Adapters/GrafanaAdapters/Functions/BuiltIn/Median.cs

```diff
@@ -43,18 +43,18 @@ public class ComputeMeasurementValue : Median<MeasurementValue>
         public override async IAsyncEnumerable<MeasurementValue> ComputeAsync(Parameters parameters, [EnumeratorCancellation] CancellationToken cancellationToken)
         {
             // Median uses immediate in-memory array load
-            MeasurementValue[] values = (await GetDataSourceValues(parameters).OrderBy(dataValue => dataValue.Value).ToArrayAsync(cancellationToken).ConfigureAwait(false)).Median();
-            int length = values.Length;
+            List<MeasurementValue> values = await GetDataSourceValues(parameters)
+                .OrderBy(dataValue => dataValue.Value)
+                .ToListAsync(cancellationToken)
+                .ConfigureAwait(false);
 
-            if (length == 0)
-                yield break;
-
-            // Median can return two values if there is an even number of values
-            MeasurementValue result = values.Last();
+            IEnumerable<MeasurementValue> median = values.Middle();
```

Comment:
Lot's of multiple enumerations even though it's two items, might as well just enumerate once:
```c#
MeasurementValue[] median = values.Middle().ToArray();
```